addons/sale/static/src/js/product/product.xml

```diff
@@ -27,49 +24,58 @@
-                />
-            </t>
-            <t t-else="">
-                <t t-call="sale.product_price" name="sale_product_configurator_optional_price"/>
-            </t>
-            <a
-                class="d-block mt-2"
-                role="button"
-                t-if="!this.props.optional &amp;&amp; !isMainProduct"
-                t-on-click="() => this.env.removeProduct(this.props.product_tmpl_id)">
-                Remove
-            </a>
-        </td>
-        <td
-            t-if="env.showQuantityAndPrice"
-            class="o_sale_product_configurator_price py-3 px-0 text-end"
-            name="price"
-        >
+        <t t-if="env.showQuantityAndPrice">
             <t t-if="!this.props.optional">
-                <t t-call="sale.product_price" name="sale_product_configurator_price"/>
+                <td
+                    class="o_sale_product_configurator_price py-3 px-0 text-end text-md-center d-block d-md-table-cell"
+                    name="price"
+                >
+                    <t t-call="sale.product_price" name="sale_product_configurator_price"/>
+                </td>
+                <td
+                    class="o_sale_product_configurator_qty py-3 px-0 text-end text-md-center d-block d-md-table-cell"
+                >
+                    <QuantityButtons
+                        quantity="this.props.quantity"
+                        setQuantity="quantity => this.env.setQuantity(this.props.product_tmpl_id, quantity)"
+                        isMinusButtonDisabled="this.props.quantity === 1 &amp;&amp; isMainProduct"
+                    />
+                    <a
+                        class="d-block mt-2 text-end"
+                        role="button"
+                        t-if="!this.props.optional &amp;&amp; !isMainProduct"
+                        t-on-click="() => this.env.removeProduct(this.props.product_tmpl_id)"
+                    >
+                        Remove
+                    </a>
+                </td>
             </t>
             <t t-else="">
-                <button
-                    name="sale_product_configurator_add_button"
-                    class="btn btn-secondary"
-                    t-att-class="{'disabled': !this.env.isPossibleCombination(this.props)}"
-                    t-on-click="() => this.env.addProduct(this.props.product_tmpl_id)"
+                <td
+                    class="o_sale_product_configurator_qty py-3 px-0 text-end text-md-center"
                 >
-                    <i class="fa fa-plus" role="img"/>
-                    <span class="ms-2 d-none d-md-inline">Add</span>
-                </button>
+                    <t t-call="sale.product_price" name="sale_product_configurator_optional_price"/>
+                </td>
+                <td
+                    class="o_sale_product_configurator_price py-3 px-0 text-end text-md-center"
+                    name="price"
```

Comment:
I'm not sure whether it's a good practice to use the same `name` attribute twice. I guess there's no harm if it doesn't cause any issues?

Reply:
They are mutually exclusive (if - else) so it will not be a problem, but this whole section should be refactored in a separate PR as you told me before IMO.